pkg/dns/hostname_generator.go

```diff
@@ -0,0 +1,39 @@
+package dns
+
+import (
+	"slices"
+	"time"
+
+	config_core "github.com/kumahq/kuma/pkg/config/core"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/hostnamegenerator/hostname"
+	"github.com/kumahq/kuma/pkg/core/runtime"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+)
+
+func SetupHostnameGenerator(rt runtime.Runtime) error {
+	if rt.GetMode() == config_core.Global {
+		return nil
+	}
+	logger := core.Log.WithName("hostnamegenerator")
+	if !slices.Contains(rt.Config().CoreResources.Enabled, "hostnamegenerators") {
+		logger.Info("HostnameGenerator is not enabled, not starting generator")
+		return nil
+	}
+	generator, err := hostname.NewGenerator(
+		logger,
+		rt.Metrics(),
+		rt.ResourceManager(),
+		100*time.Millisecond,
+		// rt.Config().IPAM.MeshService.AllocationInterval.Duration, // TODO
```

Comment:
Question: Should this interval be the same for different resources? `MeshExternalService`needs hostname generator and I think the only needed thing is to add support [here](https://github.com/kumahq/kuma/pull/10362/files#diff-4de19adee6a0c1d08e3f861fc716e75a2b540a5809e65b0a840659c1cf96b996) so it seems that we don't need a separate component for a MeshExternalService

Reply:
No you're right, we should reuse it and we shouldn't need separate configs per resource type.

Reply:
Why would we need to pass the interval of IPAM to HostnameGenerator? Do they need to run in sync?

Reply:
No, we just need _some_ interval